.github/workflows/ci.yaml

```diff
@@ -0,0 +1,97 @@
+name: For each commit and PR
+on:
+  push:
+  pull_request:
+
+jobs:
+  validation:
+    runs-on: ubuntu-latest
+    env:
+      CGO_ENABLED: 0
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.14.6'
+    - name: go fmt
+      run: go fmt ./...
+    - name: go vet
+      run: go vet ./...
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v1
+      with:
+        version: v1.30
+    - name: go vet
+      run: go test ./...
+    - run: make
+      name: Build binaries
+    - name: Upload tink-server binary
+      uses: actions/upload-artifact@v2
+      with:
+        name: tink-server
+        path: ./cmd/tink-server/tink-server
+    - name: Upload tink-cli binary
+      uses: actions/upload-artifact@v2
+      with:
+        name: tink-cli
+        path: ./cmd/tink-cli/tink-cli
+    - name: Upload tink-worker binary
+      uses: actions/upload-artifact@v2
+      with:
+        name: tink-worker
+        path: ./cmd/tink-worker/tink-worker
+    - uses: cachix/install-nix-action@v10
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - run: ./ci-checks.sh
+  docker-images:
+    env:
+      CGO_ENABLED: 0
+    runs-on: ubuntu-latest
+    needs: [validation]
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Download tink-cli binary
+      uses: actions/download-artifact@v2
+      with:
+        name: tink-cli
+    - name: Download tink-server binary
+      uses: actions/download-artifact@v2
+      with:
+        name: tink-server
+    - name: Download tink-worker binary
+      uses: actions/download-artifact@v2
+      with:
+        name: tink-worker
+    - name: quay.io/tinkerbell/tink-server
+      uses: docker/build-push-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        registry: quay.io
+        repository: tinkerbell/tink-server
+        path: ./cmd/tink-server/
+        push: ${{ startsWith(github.ref, 'refs/heads/master') }}
+        tags: latest
```

Comment:
Please add
```yaml
tag_with_ref: true
tag_with_sha: true
```
too

Reply:
Those options will push:

```
Successfully tagged quay.io/tinkerbell/tink:latest
Successfully tagged quay.io/tinkerbell/tink:pr-235-merge
Successfully tagged quay.io/tinkerbell/tink:sha-f009303
```
I don't think we want that

Reply:
I do. Maybe not `tag_wit_ref` for pushes, but definitely for tags. But I do want the sha so we can have point-in-time refs to images, and not just latest that keeps getting pushed up.

Reply:
Ok I will leave this `tag_with_sha: true`.
About tags, that's a separate discussion btw. It is not covered here and I will add it as soon as we get closer to a release lifecycle plan.

it does not look we are close to it

Reply:
Agree.